### PR TITLE
Forward Port August Patches into dev-v2.8

### PR DESCRIPTION
**Pull request by HarrisonWAffel**
Issue: https://github.com/rancher/rancher/issues/42453

This PR merges the august patches which went into 2.7 into 2.8, introducing k8s `v1.24.17`, `v1.25.13` and `v1.26.8` into dev-v2.8. This PR also brings in various ACI-CNI changes from dev-v2.7. 

Would specifically appreciate a double check of the `data.json` changes, I think they're fine but would like a second set of 👀

### CODE REVIEW

**Comment by HarrisonWAffel** [comment]
CI is now passing however I see this message in the log 

```
time="2023-09-13T16:48:10Z" level=warning msg="no aci template for k8sVersions [v1.12.9-rancher1-1 v1.13.4-rancher1-1 v1.11.1-rancher1-1 v1.13.12-rancher2-1 v1.10.5-rancher1-1 v1.11.9-rancher1-2 v1.12.5-rancher1-2 v1.15.9-rancher1-1 v1.16.15-rancher1-3 v1.16.10-rancher2-1 v1.12.0-rancher1-1 v1.15.12-rancher2-7 v1.10.11-rancher1-1 v1.15.10-rancher1-1 v1.14.10-rancher1-0 v1.15.11-rancher1-1 v1.15.11-rancher1-3 v1.13.5-rancher1-3 v1.16.15-rancher1-1 v1.12.7-rancher1-2 v1.11.3-rancher1-1 v1.16.2-rancher1-1 v1.16.6-rancher1-2 v1.15.3-rancher1-1 v1.15.4-rancher1-2 v1.15.12-rancher2-6 v1.13.5-rancher1-2 v1.15.2-rancher1-1 v1.11.9-rancher1-3 v1.15.5-rancher2-2 v1.13.5-rancher1-1 v1.8.11-rancher2-1 v1.14.1-rancher1-1 v1.14.6-rancher1-1 v1.16.6-rancher1-1 v1.14.1-rancher1-2 v1.9.5-rancher1-1 v1.11.9-rancher1-1 v1.16.15-rancher1-4 v1.13.7-rancher1-1 v1.13.10-rancher1-1 v1.12.5-rancher1-1 v1.14.5-rancher1-1 v1.12.6-rancher1-1 v1.12.3-rancher1-1 v1.13.1-rancher1-1 v1.15.5-rancher1-1 v1.15.12-rancher2-3 v1.13.9-rancher1-2 v1.13.1-rancher1-2 v1.15.12-rancher2-8 v1.15.0-rancher1-1 v1.15.6-rancher1-2 v1.16.1-rancher1-1 v1.14.8-rancher1-1 v1.16.8-rancher1-2 v1.16.10-rancher2-2 v1.11.8-rancher1-1 v1.12.6-rancher1-2 v1.13.10-rancher1-2 v1.12.1-rancher1-1 v1.14.7-rancher1-1 v1.15.11-rancher1-0 v1.10.12-rancher1-1 v1.11.5-rancher1-1 v1.13.9-rancher1-1 v1.12.10-rancher1-1 v1.14.9-rancher1-2 v1.9.7-rancher2-2 v1.16.9-rancher1-1 v1.14.3-rancher1-1 v1.12.7-rancher1-3 v1.9.7-rancher2-1 v1.12.7-rancher1-4 v1.16.13-rancher1-1 v1.10.0-rancher1-1 v1.11.2-rancher1-1 v1.11.6-rancher1-1 v1.16.7-rancher1-1 v1.15.12-rancher2-5 v1.15.11-rancher1-2 v1.10.1-rancher2-1 v1.14.8-rancher2-1 v1.16.14-rancher1-1 v1.16.8-rancher1-1 v1.12.7-rancher1-1 v1.11.2-rancher1-2 v1.15.7-rancher1-1 v1.15.12-rancher2-4 v1.16.4-rancher1-1 v1.14.9-rancher1-1 v1.15.12-rancher2-2 v1.16.13-rancher1-2 v1.12.4-rancher1-1 v1.13.12-rancher1-1 v1.10.5-rancher1-2 v1.15.5-rancher1-2 v1.13.4-rancher1-2 v1.14.10-rancher1-1 v1.16.15-rancher1-2 v1.16.2-rancher2-1 v1.15.4-rancher1-1 v1.13.11-rancher1-1 v1.16.3-rancher1-1 v1.12.10-rancher1-2 v1.10.3-rancher2-1 v1.15.12-rancher1-1] \n"
finished generating data.json
```

@mitulshah-suse Is this expected? Considering the versions are quite low I wouldn't be surprised if it is expected, but want to make sure before this gets merged

**Comment by rayandas** [comment]
@HarrisonWAffel Hi, yes this message is expected. Those versions are quite old and it occurs whenever we do a `go generate` locally for KDM.

**Comment by rayandas** [comment]
Changes looks good but need another reviewer from @rancher/rancher-team-2-hostbusters-dev

**Review by jiaqiluo — APPROVED** [review]

**Review by mitulshah-suse — APPROVED** [review]

**Pull request merged**
